source/common/router/config_impl.cc

```diff
@@ -426,17 +427,29 @@ DecoratorConstPtr RouteEntryImplBase::parseDecorator(const envoy::api::v2::Route
 }
 
 const RedirectEntry* RouteEntryImplBase::redirectEntry() const {
-  // A route for a request can exclusively be a route entry or a redirect entry.
+  // A route for a request can exclusively be a route entry, a direct response entry,
+  // or a redirect entry.
   if (isRedirect()) {
     return this;
   } else {
     return nullptr;
   }
 }
 
+const DirectResponseEntry* RouteEntryImplBase::directResponseEntry() const {
+  // A route for a request can exclusively be a route entry, a direct response entry,
+  // or a redirect entry.
+  if (isDirectResponse()) {
```

Comment:
Yeah, I think this is confusing, since a redirect entry is a direct response too